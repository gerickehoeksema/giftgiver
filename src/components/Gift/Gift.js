import React, { Component } from 'react'
import { Form, FormGroup, FormControl, FormLabel , Button } from 'react-bootstrap';

class Gift extends Component{
    constructor() {
        super()

        this.state = {
            person: '',
            present: ''
        }
    }

    updatePerson = (value) => {
        this.setState({person: value})
    }

    render(){
        return (
            <div>
                <Form>
                    <FormGroup>
                        <FormLabel >Person</FormLabel >
                        <FormControl 
                            className="person-input" 
                            onChange={event => this.updatePerson(event.target.value)}
                        />
                    </FormGroup>
                    <FormGroup>
                        <FormLabel >Present</FormLabel >
                        <FormControl 
                            className="present-input"
                            onChange={event => this.setState({present: event.target.value})}
                        />
                    </FormGroup>
                </Form>
            </div>
        )
    }
}

export default Gift