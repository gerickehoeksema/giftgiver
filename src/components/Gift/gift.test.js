import React from 'react'
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Gift from './Gift'

configure({ adapter: new Adapter() });

describe('Gift', () => {
    const gift = shallow(<Gift />)

    it('renders properly', () => {
        expect(gift.debug()).toMatchSnapshot();
    })

    it('initializes a person and a present `state`', () => {
        expect(gift.state()).toEqual({person: '', present: ''})
    })

    describe('when typing into person input', () => {
        const person = 'G'

        beforeEach(() => {
            gift.find('.person-input').simulate('change', { target: { value: person}})
        })

        it('updates the person in `state`', () => {
            expect(gift.state().person).toEqual(person)
        })
    })

    describe('when typing into person input', () => {
        const present = 'Bike'

        beforeEach(() => {
            gift.find('.present-input').simulate('change', {target: {value: present}})
        })

        it('updates the present in `state`',() => {
            expect(gift.state().present).toEqual(present)
        })
    })
})