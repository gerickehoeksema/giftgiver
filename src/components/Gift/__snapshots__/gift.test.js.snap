// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gift renders properly 1`] = `
"<div>
  <Form inline={false}>
    <FormGroup>
      <FormLabel column={false} srOnly={false}>
        Person
      </FormLabel>
      <FormControl className=\\"person-input\\" onChange={[Function: onChange]} />
    </FormGroup>
    <FormGroup>
      <FormLabel column={false} srOnly={false}>
        Present
      </FormLabel>
      <FormControl className=\\"present-input\\" onChange={[Function: onChange]} />
    </FormGroup>
  </Form>
</div>"
`;
